### PR TITLE
Add crypto-js w/ git auto-update & update package.json

### PR DESCRIPTION
**Pull request by evanvosberg**
Pull request for issue: #9958
Related issue(s): https://github.com/brix/crypto-js/issues/75

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * License and its reference:
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 94
   - Count of stars: 1591
   - Count of forks: 273
 * NPM download stats (optional):
   - Downloads in the last day: 16.936
   - Downloads in the last week: 92.935
   - Downloads in the last month: 390.481

# Essential checklist
 * [x] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by evanvosberg** [comment]
Would like to add the auto update before merge ... so please wait for.

**Comment by evanvosberg** [comment]
Ok looks good now.

File: ajax/libs/crypto-js/package.json [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
+    "url": "https://github.com/brix/crypto-js.git"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/brix/crypto-js.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "**/*"
```
Comment (pvnr0082t) [inline ajax/libs/crypto-js/package.json]: @evanvosberg It's a little bit not safe or no sense to use **/* under a non-dist folder. Let's use `"*.+(js|css|map)"` instead.

File: ajax/libs/crypto-js/package.json [hunk 2]
```diff
@@ -1,29 +1,49 @@
...
-    "url": "http://crypto-js.googlecode.com/svn"
+  "filename": "crypto-js.js",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/brix/crypto-js.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "**/*.+(js|json|map)"
```
Comment (pvnr0082t) [inline ajax/libs/crypto-js/package.json]: @evanvosberg CDNJS does not support `json` files yet, let's remove that. Thanks.

**Comment by maruilian11** [comment]
hi @evanvosberg, could you please only commit the package.json and remove other files, which will help us review your PR more quickly.
And please remove `version` field in package.json if you only commit it.
Many thanks!

**Comment by PeterDaveHello** [comment]
@maruilian11 your request is odd, that's no need at all.

**Comment by evanvosberg** [comment]
The releases of crypro-js don't include minified files and the releases are already out for a longer time like on npm or bower. So I can remove the files and just update the package.json ... pretty convenience for me + I would change the build process for future versions to include minified files.

But if all the existing versions need to be minified as well I would need to do this manually ...

File: ajax/libs/crypto-js/package.json (hunk 2 above)
Comment (PeterDaveHello) [inline ajax/libs/crypto-js/package.json]: If the json files are optional and useful, we can still add it, we'll support it in future, but currently will not work.

**Comment by PeterDaveHello** [comment]
@evanvosberg for the old versions, usually will be minified by my side, so that's not a problem.

**Comment by evanvosberg** [comment]
@PeterDaveHello so anything more to do from my side except adjusting the `files: "**/*.+(js|json|map)"`?

**Comment by PeterDaveHello** [comment]
I'll wait for other reviewers' review first, I'm usually the last stage of the review 😄

**Comment by pvnr0082t** [comment]
@evanvosberg So would you like to remove the files and just update `package.json`, or you want to keep them? If you want to keep them, please manually minify them, thanks. :+1:

**Comment by evanvosberg** [comment]
@pvnr0082t as @PeterDaveHello commented there no requirement to remove the files and that it's not minified is also not a problem. The package.json is already updated.

**Comment by PeterDaveHello** [comment]
@evanvosberg Oops, if you do not provide minified files, it will be a problem, unless you do not commit the files, sorry for the misunderstanding.

File: ajax/libs/crypto-js/3.1.9/x64-core.min.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+!function(t,e){"object"==typeof exports?module.exports=exports=e(require("./core.min")):"function"==typeof define&&define.amd?define(["./core.min"],e):e(t.CryptoJS)}(this,function(t){return function(e){var i=t,n=i.lib,o=n.Base,r=n.WordArray,s=i.x64={};s.Word=o.extend({init:function(t,e){this.high=t,this.low=e}}),s.WordArray=o.extend({init:function(t,i){t=this.words=t||[],i!=e?this.sigBytes=i:this.sigBytes=8*t.length},toX32:function(){for(var t=this.words,e=t.length,i=[],n=0;n<e;n++){var o=t[n];i.push(o.high),i.push(o.low)}return r.create(i,this.sigBytes)},clone:function(){for(var t=o.clone.call(this),e=t.words=this.words.slice(0),i=e.length,n=0;n<i;n++)e[n]=e[n].clone();return t}})}(),t});
+//# sourceMappingURL=ajax/libs/crypto-js/3.1.9/x64-core.min.js.map
```
Comment (pvnr0082t) [inline ajax/libs/crypto-js/3.1.9/x64-core.min.js]: I think this sourceMappingURL should be `//# sourceMappingURL=./x64-core.min.js.map`. 
@PeterDaveHello I remember we met this problem before, right?

Comment (PeterDaveHello) [inline ajax/libs/crypto-js/3.1.9/x64-core.min.js]: That may be caused by wrong config.

**Comment by PeterDaveHello** [comment]
@pvnr0082t if it looks for you please help squash and rebase before ping me next time.

**Comment by maruilian11** [comment]
@evanvosberg Since you have added 3.1.2, do we also need to host `release-3.1.2` ~ `release-3.1.2-5`? Thanks!

**Comment by maruilian11** [comment]
@PeterDaveHello tag `3.1.9-1` just released few days agp, do we also need to keep the latest version in this PR? in this case?

**Comment by PeterDaveHello** [comment]
@maruilian11 explained before, unless the auto-update config is out-date, don't make the process more complex ...

**Comment by evanvosberg** [comment]
@maruilian11 All current versions are in the pr, the autoupdate should apply for new versions in the future only.

**Pull request merged**